presto-main/src/main/java/com/facebook/presto/sql/analyzer/MetadataExtractor.java

```diff
@@ -186,7 +197,9 @@ protected Void visitTable(Table table, MetadataExtractorContext context)
             }
 
             // This could be either tableName, view, or MView
-            context.addTable(tableName);
+            if (!context.tableExistsInCommonTableExpressionNames(new Identifier(tableName.getObjectName()))) {
```

Comment:
Similar to below comment, this could be generalized by checking context.shouldSkipProcessing(identifier)?

Reply:
sounds good